### PR TITLE
[RFC][PL-355] Create php logger

### PR DESCRIPTION
**Pull request by raffo0707**
https://uniplaces.atlassian.net/browse/PL-355

### CODE REVIEW

File: src/Monolog/Processors/CommonProcessor.php [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+
+    /**
+     * @param array $record
+     *
+     * @return array
+     */
+    protected function buildCommon(array $record): array
+    {
+        $commonFields = [
+            'time' => Carbon::now()->toRfc3339String(),
```
Comment (tlfbrito) [inline src/Monolog/Processors/CommonProcessor.php]: I don't think we need to depend on carbon to do this
```php
$now = new \Datetime();
var_dump($now->format(\DateTime::RFC3339));
var_dump(time());
var_dump(date(DATE_RFC3339));
```
```
string(25) "2017-12-06T17:03:05+01:00"
int(1512576185)
string(25) "2017-12-06T17:03:05+01:00"
```

**Review by tlfbrito — COMMENTED** [review]
I would change the repository name to something like:

* uniplaces-monolog
* uniplaces-monolog-extension
* uni-monolog
* uni-monolog-extra

File: composer.json [hunk 2]
```diff
@@ -0,0 +1,17 @@
+{
+    "name": "uniplaces/php-logger",
+    "description": "logger processors for uniplaces standards",
+    "type": "library",
+    "require": {
+        "php": ">=7.0.0",
+        "nesbot/carbon": "^1.22",
```
Comment (tlfbrito) [inline composer.json]: remove carbon dependency

File: composer.json [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+        "symfony/http-foundation": "^2.8|^3.4|^4.0"
+    },
+    "require-dev": {
+        "phpunit/phpunit": "~6.0"
```
Comment (tlfbrito) [inline composer.json]: "^6.5"

File: composer.json [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    },
+    "autoload": {
+        "psr-4": { "phpLogger\\": "src/" }
```
Comment (tlfbrito) [inline composer.json]: `"Uniplaces\\": "src"`

File: composer.json [hunk 5]
```diff
@@ -0,0 +1,17 @@
+{
+    "name": "uniplaces/php-logger",
+    "description": "logger processors for uniplaces standards",
+    "type": "library",
+    "require": {
+        "php": ">=7.0.0",
```
Comment (tlfbrito) [inline composer.json]: `"^7.0"`

**Review by tlfbrito — APPROVED** [review]
Reviewed

File: src/Monolog/Processors/CommonProcessor.php [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+    {
+        $commonFields = [
+            'time' => time(),
+            'app-id' => $this->appId,
+            'env' => $this->env,
+            'msg' => $record['message'],
+            'level' => strtolower($record['level_name']),
+            'git-hash' => $this->gitHash
+        ];
+
```
Comment (jkrnak) [inline src/Monolog/Processors/CommonProcessor.php:77]: one to many new lines #minor

File: tests/Monolog/Processors/CommonProcessorTest.php [hunk 7]
```diff
@@ -0,0 +1,86 @@
+<?php
+
+namespace tests\Monolog\Processors;
+
+use Symfony\Component\HttpFoundation\Request;
```
Comment (jkrnak) [inline tests/Monolog/Processors/CommonProcessorTest.php]: Please do an optimize imports in PHPUnit, it organises imports alphabetically.

File: src/Monolog/Processors/CommonProcessor.php [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+     * @return array
+     */
+    protected function buildCommon(array $record): array
+    {
+        $commonFields = [
+            'time' => time(),
+            'app-id' => $this->appId,
+            'env' => $this->env,
+            'msg' => $record['message'],
+            'level' => strtolower($record['level_name']),
```
Comment (jkrnak) [inline src/Monolog/Processors/CommonProcessor.php:74]: Test line 46 checks if `level === 0` 
I'm probably missing something, but how can it be `0` if it's a lowercase version of the name?

File: composer.json [hunk 9]
```diff
@@ -0,0 +1,16 @@
+{
+    "name": "uniplaces/php-logger",
```
Comment (jkrnak) [inline composer.json]: Please change the package name too.

File: README.md [hunk 10]
```diff
@@ -1 +1 @@
-# php-logger
\ No newline at end of file
+# phpLogger
```
Comment (jkrnak) [inline README.md]: Please add a bit more readme :)

**Review by jkrnak — CHANGES_REQUESTED** [review]

File: tests/Monolog/Processors/CommonProcessorTest.php (hunk 7 above)
Comment (tlfbrito) [reply tests/Monolog/Processors/CommonProcessorTest.php]: in PHPStorm, right @jkrnak ?

**Review by jkrnak — APPROVED** [review]

**Pull request merged**